nixos/tests/cpython-gdb.nix

```diff
@@ -0,0 +1,32 @@
+import ./make-test-python.nix ({ pkgs, ... }:
```

Comment:
Is such an expensive NixOS test needed? Could this not be done in just a normal build and then added to the `passthru.tests` of the interpreters? See `pkgs/development/interpreters/python/tests.nix`

Reply:
Interesting. I had not discovered those tests. The reason I initially gravitated towards nixos tests is firstly I wasn't sure whether ptrace would work inside nix's sandbox, secondly I thought "at least we're only running it once", and of course only on a platform I know it should work for (`separateDebugInfo` isn't a thing on e.g. macos AFAIK).

I'll take a look at whether adding it to `integrationTests` in that file works out. Of course it will have to restrict itself to cpython & linux..

Reply:
@FRidh quite right, these work fine as integration tests. Updated.